client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -632,16 +632,20 @@ function mediaButton( editor ) {
 
 		gallery = assign( {}, MediaConstants.GalleryDefaultAttrs, gallery.attrs.named );
 
-		gallery.items = gallery.ids.split( ',' ).map( ( id ) => {
-			id = parseInt( id, 10 );
+		gallery.items = [];
 
-			const media = MediaStore.get( selectedSite.ID, id );
-			if ( ! media ) {
-				MediaActions.fetch( selectedSite.ID, id );
-			}
+		if ( gallery.ids ) {
+			gallery.items = gallery.ids.split( ',' ).map( ( id ) => {
+				id = parseInt( id, 10 );
```

Comment:
@aduth Hadn't considered non-numeric values! Empty strings are falsey, so I don't believe that would've be an issue, but this does appear to have more coverage for unexpected cases than my solution. Thanks!